### PR TITLE
feat: add parallax background section to the home page

### PR DESCRIPTION
**Pull request by tetianaman**
https://github.com/user-attachments/assets/d99c4b28-5aa0-4773-8252-0e0a304c051e

### CODE REVIEW

File: app/routes/_index/index.module.scss [hunk 1]
```diff
@@ -38,3 +38,7 @@
     font-size: clamp(60px, 6vw, 110px);
     text-transform: uppercase;
 }
+
+.textOverlayCardLinkLabel {
```
Comment (tetianaman) [inline app/routes/_index/index.module.scss:42]: `LabelWithArrow` `.root` has `width: 220px` and its scss file is imported after _common.scss_, so I cannot override it from _common.scss_ style.

File: src/components/background-parallax/background-parallax.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import classNames from 'classnames';
```
Comment (tetianaman) [inline src/components/background-parallax/background-parallax.tsx:1]: I'll move this code back to _visual-effects/index.tsx_

File: src/styles/common.scss [hunk 3]
```diff
@@ -103,3 +103,52 @@
     font: var(--heading4);
     text-align: center;
 }
+
+.backgroundParallax {
+    margin: 0 -2%;
+    width: 104%; // This is awful
```
Comment (tetianaman) [inline src/styles/common.scss:109]: thinking how to fix this normally

File: src/styles/index.scss [hunk 4]
```diff
@@ -17,8 +15,6 @@
 
 body {
     margin: 0;
-    place-items: center;
-    justify-content: center;
```
Comment (alisey) [inline src/styles/index.scss:21]: Remove this change from this PR.

File: src/styles/common.scss [hunk 5]
```diff
@@ -103,3 +103,52 @@
     font: var(--heading4);
     text-align: center;
 }
+
+.backgroundParallax {
```
Comment (alisey) [inline src/styles/common.scss:107]: This doesn't make sense as a common style.

File: src/components/visual-effects/background-parallax.tsx [hunk 6]
```diff
@@ -11,6 +11,8 @@ export interface BackgroundParallaxProps extends HTMLAttributes<HTMLDivElement>
      *   the viewport, similar to `background-attachment: fixed`.
      */
     parallaxStrength?: number;
+    imageUrl?: string;
+    className?: string;
```
Comment (alisey) [inline src/components/visual-effects/background-parallax.tsx:15]: You don't need `className`

File: src/components/visual-effects/background-parallax.tsx [hunk 7]
```diff
@@ -11,6 +11,8 @@ export interface BackgroundParallaxProps extends HTMLAttributes<HTMLDivElement>
      *   the viewport, similar to `background-attachment: fixed`.
      */
     parallaxStrength?: number;
+    imageUrl?: string;
```
Comment (alisey) [inline src/components/visual-effects/background-parallax.tsx:14]: `backgroundImageUrl`

File: src/components/visual-effects/background-parallax.tsx [hunk 8]
```diff
@@ -57,13 +61,15 @@ export const BackgroundParallax: FC<BackgroundParallaxProps> = ({
...
             ref={ref}
+            className={className}
             style={{
                 backgroundAttachment: 'fixed',
                 // With `background-attachment: fixed`, the `cover` value for
                 // background size is calculated relative to the viewport.
                 backgroundSize: 'cover',
                 backgroundRepeat: 'no-repeat',
                 backgroundPositionX: 'center',
+                backgroundImage: `url("${imageUrl}")`,
```
Comment (alisey) [inline src/components/visual-effects/background-parallax.tsx:72]: Don't add this if it's undefined.

File: app/routes/_index/index.module.scss [hunk 9]
```diff
@@ -38,3 +38,15 @@
...
+.floatingCardBackground {
+    width: 100vw;
+    margin-left: -2%;
+    padding: 7.8% 11.7%;
+    aspect-ratio: 2;
+}
+
+.floatingCardLinkLabel {
+    width: 170px;
+    height: 34px;
```
Comment (alisey) [inline app/routes/_index/index.module.scss:51]: Why fixed dimensions? Why make them update it every time they want to change the text?

File: app/routes/_index/index.module.scss [hunk 10]
```diff
@@ -38,3 +38,15 @@
     font-size: clamp(60px, 6vw, 110px);
     text-transform: uppercase;
 }
+
+.floatingCardBackground {
+    width: 100vw;
+    margin-left: -2%;
```
Comment (alisey) [inline app/routes/_index/index.module.scss:44]: Why only left margin?

File: app/routes/_index/route.tsx [hunk 11]
```diff
@@ -56,6 +57,28 @@ export default function HomePage() {
                     <div className="linkCardTitle">On the Go</div>
                 </CategoryLink>
             </div>
+
+            <BackgroundParallax
+                parallaxStrength={0.75}
```
Comment (alisey) [inline app/routes/_index/route.tsx:62]: > parallaxStrength={0.75}

It's the default value, don't need to specify it.

File: src/components/visual-effects/background-parallax.tsx (hunk 6 above)
Comment (tetianaman) [reply src/components/visual-effects/background-parallax.tsx:15]: I don't want to add additional styles to the inline style:
```
.floatingCardBackground {
    width: 100vw;
    margin-left: -2%;
    padding: 7.8% 11.7%;
    aspect-ratio: 2;
}
```

File: app/routes/_index/index.module.scss (hunk 9 above)
Comment (tetianaman) [reply app/routes/_index/index.module.scss:51]: That's a tricky problem to make `LabelWithArrow` width dynamic

File: src/components/visual-effects/background-parallax.tsx (hunk 6 above)
Comment (alisey) [reply src/components/visual-effects/background-parallax.tsx:15]: You don't need the className prop, this component already supports `className`.

interface BackgroundParallaxProps **`extends HTMLAttributes<HTMLDivElement>`**

File: src/components/visual-effects/background-parallax.tsx [hunk 12]
```diff
@@ -57,13 +61,15 @@ export const BackgroundParallax: FC<BackgroundParallaxProps> = ({
...
             ref={ref}
+            className={className}
             style={{
                 backgroundAttachment: 'fixed',
                 // With `background-attachment: fixed`, the `cover` value for
                 // background size is calculated relative to the viewport.
                 backgroundSize: 'cover',
                 backgroundRepeat: 'no-repeat',
                 backgroundPositionX: 'center',
+                ...(backgroundImageUrl ? { backgroundImage: `url("${backgroundImageUrl}")` } : {}),
```
Comment (alisey) [inline src/components/visual-effects/background-parallax.tsx:72]:
```ts
backgroundImage: backgroundImageUrl ? `url("${backgroundImageUrl}")` : undefined,
```

File: app/routes/_index/index.module.scss (hunk 9 above)
Comment (alisey) [reply app/routes/_index/index.module.scss:51]: What problem?

File: src/components/visual-effects/background-parallax.tsx (hunk 12 above)
Comment (alisey) [reply src/components/visual-effects/background-parallax.tsx]: ?

File: app/routes/_index/index.module.scss (hunk 9 above)
Comment (tetianaman) [reply app/routes/_index/index.module.scss:51]: Oh, seems I found the solution 🥳 

[https://jsfiddle.net/6fz4yevk/14/](https://jsfiddle.net/6fz4yevk/14/)

Comment (alisey) [reply app/routes/_index/index.module.scss:51]: https://jsfiddle.net/uc9wvLym/

1. Use aria-hidden.
2. Don't put the hidden element inside of visible.
3. Transition should be defined on the normal state, not on the hover state.

**Review by alisey — APPROVED** [review]

**Pull request merged**